src/modules/mc_pos_control/McPosControl.cpp

```diff
@@ -0,0 +1,1955 @@
+
+#include <px4_config.h>
+#include <px4_defines.h>
+#include <px4_tasks.h>
+#include <px4_posix.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <math.h>
+#include <arch/board/board.h>
+
+
+
+#include <float.h>
+#include <systemlib/mavlink_log.h>
+#include <platforms/px4_defines.h>
+
+#include "McPosControl.hpp"
```

Comment:
@dagar: the mc_pos_control_test is anyway a hack, because it tests only one method, and I think the only reason why this method is in the public scope is because otherwise it is difficult to test it. Therefore, I wouldnt worry too much about this particular unit test, but rather I am planing on doing first tests against the class interface and then I will start putting methods, such as the cross_sphere, in different classes where again I can write tests for it to ensure nothing breaks. But I think it is the wrong approach to make a method class public just because the structure doesnt allow testing it if it is private.
About syntax: I really would like to have a semantic / syntax px4 guide (for instance google code style) and put it on the dev guide. I also think we could decide on how to start and close an app and then put a final version on the dev guide as well, since all apps follow a similar structure anyway.